src/db/transformCSV.js

```diff
@@ -0,0 +1,143 @@
+//   output: process.stdout,
+//   terminal: false,
+// });
+
+// exports.readQuestions = async () => {
+//   let headers = [];
+//   let row = {};
+
+//   const addAllQuestions = async (row) => {
+//     User.queryGetUser(row.asker_name)
+//       .then(([{ id }]) => id)
+//       .then((id) => {
+//         const question = {
+//           id: row.id,
+//           body: row.body,
+//           reported: row.reported,
+//           helpfulness: row.helpful,
+//           usernameId: id,
+//           productId: row.product_id,
+//           createdAt: row.date_written,
+//         };
+//         return question;
+//       })
+//       .then((question) => Question.queryAddQuestion(question))
+//       .catch((err) => console.log(err.detail));
+//   };
+
+//   questionsCSV.forEach((line, index) => {
+//     if (index === 0) {
+//       headers = line.split(', ').map((header) => header);
+//       return;
+//     }
+
+//     row = line
+//       .split(',')
+//       .reduce((acc, value, idx) => {
+//         if (Number(value)) {
+//           return { ...acc, [headers[idx]]: Number(value) };
+//         }
+//         if (headers[idx] === 'reported') {
+//           return { ...acc, [headers[idx]]: Boolean(value) };
+//         }
+//         return { ...acc, [headers[idx]]: value.slice(1, -1) };
+//       }, {});
+
+//     User.queryAddUser({ username: row.asker_name, email: row.asker_email })
+//       .catch((err) => console.log(err.detail))
+//   });
+// };
+
+// questionsCSV.on('line', (line) => {
+//   counter += 1;
+//   if (counter === 1) {
+//     headers = line.split(', ');
+//   } else {
+//     const data = line.split(',');
+//     row = headers.reduce((acc, header, index) => {
+//       if (header === 'id' || header === 'product_id' || header === 'helpful') {
+//         return { ...acc, [header]: Number(data[index]) };
+//       }
+
+//       if (header === 'reported') {
+//         return { ...acc, [header]: Boolean(data[index]) };
+//       }
+
+//       return { ...acc, [header]: data[index].slice(1, -1) };
+//     }, {});
+
+//     const user = {
+//       username: row.asker_name,
+//       email: row.asker_email,
+//     };
+
+//     User.queryAddUser(user)
+//       .then((response) => console.log('I inserted this user: ', response))
+//       .catch((err) => console.log(err.detail));
+
+//     User.queryGetUser(user.username)
+//       .then(([{ id }]) => {
+//         const question = {
+//           id: row.id,
+//           body: row.body,
+//           reported: row.reported,
+//           helpfulness: row.helpful,
+//           usernameId: id,
+//           productId: row.product_id,
+//           createdAt: row.date_written,
+//         };
+//         Question.queryAddQuestion(question)
+//           .then((response) => console.log(response))
+//           .catch((err) => console.log(err));
+//       });
+//   }
+// });
+
+// exports.readQuestions();
```

Comment:
Lots of commented out code. Delete if you don't need this anymore. If you're leaving it for reference or future use, perhaps leave a comment that says so.

Reply:
Thanks. I will clean up code in future PRs.